feat: add `joinEnv` and `joinUrl` string functions and 2 new system vars

solvingj opened the pull request:
Add two new system variables which map directly to platform specific go constants:
  - PATH_LIST_SEPARATOR -> os.PathListSeparator
  - FILE_PATH_SEPARATOR -> os.PathSeparator

Add joinEnv string function which is sibling to joinPath:
- strings.Join(elem, string(os.PathListSeparator))

Add joinUrl string function which is sibling to joinPath:
- path.Join(elem...)

Add all to docs

closes #2406

solvingj committed:
Add two new special vars, which map directly to platform specific go constants:
  - PATH_LIST_SEPARATOR -> os.PathListSeparator
  - FILE_PATH_SEPARATOR -> os.PathSeparator

Add joinEnv string function which is sibling to joinPath:
- strings.Join(elem, string(os.PathListSeparator))

Add joinUrl string function which is sibling to joinPath:
- path.Join(elem...)

Add all to docs

closes #2406

solvingj committed:
Remove accidental guide.md doc change

solvingj committed:
Remove accidental guide.md doc change2

trulede reviewed (commented):
@solvingj there are quite a few functions already in https://go-task.github.io/slim-sprig/ and perhaps its better to use those existing list functions (join, append, prepend) since that gives a better control over how such lists/strings are generated.

The PathSeparator and PathListSeparator are useful however I would suggest following the existing most-predominate naming convention:

* PathSeparator -> osPathSeparator (note also the absence of "file")
* PathListSeparator -> osPathListSeparator 

since the allcaps just looks ODD.


Observation: this could also be done with a .env file, introducing these constants (and others). Perhaps this PR is more effort than value.

vmaerten reviewed (changes requested):
Hello!
Sorry for the delay, we have been busy lately.

`PATH_LIST_SEPARATOR` / `FILE_PATH_SEPARATOR` does not work in the templating system.

If you take this Taskfile : 
```yml
# https://taskfile.dev

version: '3'

tasks:
  default:
    cmds:
      - echo "{{.PATH_LIST_SEPARATOR}}"
      - echo "{{.FILE_PATH_SEPARATOR}}"

```
 You will get this result: 

<img width="224" height="80" alt="Image" src="https://github.com/user-attachments/assets/ffddefa0-b4ca-44cb-8923-4e5413ec3600" />

For these variables, you need to define them in `getSpecialVars` inside `compiler.go`

solvingj committed:
Move to compiler.go per feedback from @vmaerten

solvingj commented:
> For these variables, you need to define them in `getSpecialVars` inside `compiler.go`

Ah, of course, moved...

andreynering commented:
Hi @solvingj,

Looks like this build is broken. This needs a rebase.

solvingj committed:
Fix merge conflicts with main and add special vars to summary filter

- Rebased on latest main branch (origin/main)
- Resolved conflicts in compiler.go: merged filepath.ToSlash() changes with new special variables
- Fixed type conversion for os.PathListSeparator and os.PathSeparator (rune to string)
- Added PATH_LIST_SEPARATOR and FILE_PATH_SEPARATOR to summary filter to prevent them from appearing in task summaries (consistent with other special variables)
- All tests passing including linting

solvingj commented:
@andreynering done. All tests pass locally now.

solvingj commented:
@andreynering any additional blockers?

andreynering committed:
docs: make some documentation adjustments

andreynering committed:
Merge branch 'main' into add_joinenv

andreynering reviewed (approved):
Thank you for your patience!
